abad.go

```diff
@@ -97,4 +121,103 @@ func (a *Abad) evalUnaryExpr(expr *ast.UnaryExpr) (Obj, error) {
 	}
 
 	return num, nil
+}
+
+func (a *Abad) evalExpr(n ast.Node) (types.Value, error) {
+	if !ast.IsExpr(n) {
+		panic("internal error: not an expression")
+	}
+
+	switch n.Type() {
+	case ast.NodeNumber:
+		val := n.(ast.Number)
+		return types.Number(val.Value()), nil
+	case ast.NodeIdent:
+		val := n.(ast.Ident)
+		return a.evalIdentExpr(val)
+	case ast.NodeMemberExpr:
+		val := n.(*ast.MemberExpr)
+		return a.evalMemberExpr(val)
+	case ast.NodeCallExpr:
+		val := n.(*ast.CallExpr)
+		return a.evalCallExpr(val)
+	case ast.NodeUnaryExpr:
+		expr := n.(*ast.UnaryExpr)
+		return a.evalUnaryExpr(expr)
+	}
+
+	panic("unreachable")
+	return nil, nil
+}
+
+func (a *Abad) evalIdentExpr(ident ast.Ident) (types.Value, error) {
+	val, err := a.global.Get(utf16.Str(ident))
+	if err != nil {
+		return nil, err
+	}
+
+	if types.StrictEqual(val, types.Undefined) {
+		return nil, fmt.Errorf("%s is not defined",
+			ident.String())
+	}
+
+	return val, nil
+}
+
+func (a *Abad) evalMemberExpr(member *ast.MemberExpr) (types.Value, error) {
+	objval, err := a.evalExpr(member.Object)
+	if err != nil {
+		return nil, err
+	}
+
+	if objval.Kind() != types.KindObject {
+		panic("wrapping primitive values not implemented yet")
+	}
+
+	obj, err := objval.ToObject()
+	if err != nil {
+		return nil, err
+	}
+
+	return obj.Get(utf16.Str(member.Property))
```

Comment:
Should'nt these fields already be utf16.Str ? I'm starting to get some bad feelings of lack of uniformity and casting around =/ (not the first time I see this kind of cast).

Reply:
MemberExpr.Property is of type Ident. Ident is an alias to utf16.Str. Here, it's just a conversion, not a cast.

Reply:
Hmm I got confused thinking that utf.Str was the function that takes a string and returns the utf16.Str thing, sorry =)